pkg/controller/hibernation/ibmcloud_actuator.go

```diff
@@ -0,0 +1,173 @@
+	}
+
+	return nil
+}
+
+// StartMachines will select machines belonging to the given ClusterDeployment
+func (a *ibmCloudActuator) StartMachines(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) error {
+	logger = logger.WithField("cloud", "ibmcloud")
+	ibmCloudClient, err := a.ibmCloudClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return err
+	}
+
+	instances, err := getIBMCloudClusterInstances(cd, ibmCloudClient, stoppedOrStoppingStates, logger)
+	if err != nil {
+		return err
+	}
+	if len(instances) == 0 {
+		logger.Info("No instances were found to start")
+		return nil
+	}
+	err = ibmCloudClient.StartInstances(instances)
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to start IBM Cloud instances")
+		return err
+	}
+
+	return nil
+}
+
+func ibmCloudInstanceNames(instances []vpcv1.Instance) []string {
+	names := make([]string, len(instances))
+	for i, instance := range instances {
+		names[i] = *instance.Name
+	}
+	return names
+}
+
+// MachinesRunning will return true if the machines associated with the given
+// ClusterDeployment are in a running state. It also returns a list of machines that
+// are not running.
+func (a *ibmCloudActuator) MachinesRunning(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) (bool, []string, error) {
+	logger = logger.WithField("cloud", "ibmcloud")
+	logger.Infof("checking whether machines are running")
+	ibmCloudClient, err := a.ibmCloudClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return false, nil, err
+	}
+	instances, err := getIBMCloudClusterInstances(cd, ibmCloudClient, notRunningStates, logger)
+	if err != nil {
+		return false, nil, err
+	}
+	return len(instances) == 0, ibmCloudInstanceNames(instances), nil
+}
+
+// MachinesStopped will return true if the machines associated with the given
+// ClusterDeployment are in a stopped state. It also returns a list of machines
+// that have not stopped.
+func (a *ibmCloudActuator) MachinesStopped(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) (bool, []string, error) {
+	logger = logger.WithField("cloud", "ibmcloud")
+	logger.Infof("checking whether machines are stopped")
+	ibmCloudClient, err := a.ibmCloudClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return false, nil, err
+	}
+	instances, err := getIBMCloudClusterInstances(cd, ibmCloudClient, notStoppedStates, logger)
+	if err != nil {
+		return false, nil, err
+	}
+	return len(instances) == 0, ibmCloudInstanceNames(instances), nil
+}
+
+func getIBMCloudClient(cd *hivev1.ClusterDeployment, c client.Client, logger log.FieldLogger) (ibmclient.API, error) {
+	secret := &corev1.Secret{}
+	err := c.Get(context.TODO(), client.ObjectKey{Name: cd.Spec.Platform.IBMCloud.CredentialsSecretRef.Name, Namespace: cd.Namespace}, secret)
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to fetch IBM Cloud credentials secret")
```

Comment:
curious why this isn't just
```suggestion
		logger.WithError(err).Error("failed to fetch IBM Cloud credentials secret")
```